dex/farm-staking/src/lib.rs

```diff
@@ -182,13 +175,64 @@ pub trait Farm:
         #[payment_nonce] token_nonce: Nonce,
         #[payment_amount] amount: BigUint,
         #[var_args] opt_accept_funds_func: OptionalArg<ManagedBuffer>,
+    ) -> SCResult<ExitFarmResultType<Self::Api>> {
+        self.unstake_farm_common(
+            payment_token_id,
+            token_nonce,
+            amount,
+            opt_accept_funds_func,
+            None,
+        )
+    }
+
+    #[payable("*")]
+    #[endpoint(unstakeFarmThroughProxy)]
+    fn unstake_farm_through_proxy(
+        &self,
+        #[payment_multi] payments: ManagedVec<EsdtTokenPayment<Self::Api>>,
+    ) -> SCResult<ExitFarmResultType<Self::Api>> {
+        let caller = self.blockchain().get_caller();
+        self.require_whitelisted(&caller);
+
+        require!(payments.len() == 2, "Invalid payments amount");
+
+        let first_payment = payments.get(0).unwrap();
```

Comment:
add a comment that first token is added into the RIDE liquidity and will be taken out when user unBonds.